lib/matplotlib/lines.py

```diff
@@ -472,7 +472,7 @@ def contains(self, mouseevent):
 
         # Convert pick radius from points to pixels
         if self.figure is None:
-            warnings.warn('no figure set when check if mouse is on line')
+            logging.warning('no figure set when check if mouse is on line')
```

Comment:
I think warn_external here (can be callback or direct call).

Reply:
Isn't this gui interaction?

Reply:
I was thinking that one can also call contains() directly, but actually that's almost certainly much rarer, so logging is fine.